hivemind/client/moe.py

```diff
@@ -75,10 +74,10 @@ def forward(self, input: torch.Tensor, *args: torch.Tensor, **kwargs: torch.Tens
             input_for_gating = input
 
         # 1. compute scores and find most appropriate experts with beam search
-        grid_scores = self.proj(input_for_gating).split_with_sizes(self.grid_size, dim=-1)
+        grid_scores = self.proj(input_for_gating).split_with_sizes(self.beam_search.grid_size, dim=-1)
 
-        chosen_experts: List[List[RemoteExpert]] = self.dht.batch_find_best_experts(
-            self.uid_prefix, [scores.detach().cpu().numpy() for scores in grid_scores], self.k_best, **self.dht_kwargs)
+        chosen_experts: List[List[RemoteExpert]] = self.beam_search.batch_find_best_experts(
```

Comment:
Do we need typing annotations for this method? We can add it to its return value

Reply:
[same as originally] i would argue that it simplifies reading the complex code of moe.py